packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -495,6 +688,11 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     if (!_acceptedActivePointers.remove(pointer)) {
       resolvePointer(pointer, GestureDisposition.rejected);
     }
+
+    if (_activePointer == pointer) {
+      _activePointer =
+        _acceptedActivePointers.isNotEmpty ? _acceptedActivePointers.first : null;
```

Comment:
Refer to Android [RecyclerView](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-master-dev/recyclerview/recyclerview/src/main/java/androidx/recyclerview/widget/RecyclerView.java)

![image](https://github.com/flutter/flutter/assets/61075224/34381d0c-02c6-43c1-8b9d-b4f2a2a101e4)